listenbrainz_spark/recommendations/candidate_sets.py

```diff
@@ -149,8 +149,20 @@ def get_top_artists(mapped_listens_subset, top_artist_limit, users):
                                                          'user_name',
                                                          'total_count') \
                                                  .where(top_artist_df.user_name.isin(users))
+        try:
```

Comment:
why are we doing this? are we trying to check if the top_artist_df is not empty? is there a more obvious way to do this? We should extract this into a function at least.

Reply:
Yes. I am not sure about any other way. Can dump this into a different function though.

Reply:
Can we just make the is empty dataframe function return a book and use that, instead of all the try excepts?